cognite/powerops/client/shop_result_files.py

```diff
@@ -89,6 +93,64 @@ def _parse_file(self, file: TextIO) -> dict:
     def file_content(self) -> str:
         return yaml.safe_dump(self.data, sort_keys=False)
 
+    def _prepare_plot_time_series(self, keys: Union[str, Sequence[str]]) -> dict:
+        if isinstance(keys, str):
+            keys = [keys]
+        return {key: self._retrieve_time_series_dict(key) for key in keys}
+
+    def _retrieve_time_series_dict(self, keys: str) -> dict[datetime, float]:
+        try:
+            data = self[keys]
+            assert isinstance(data, dict)
+            assert all(isinstance(key, datetime) for key in data.keys())
+            assert all(isinstance(value, (float, int)) for value in data.values())
+            return data
+
+        except KeyError:
+            logger.error(f'Key "{keys}" not found in {self.name}')
+        except AssertionError:
+            logger.error("Data cannot be plotted as a a time series")
+        return {}
+
+    def list_model_time_series_keys(
+        self,
+        matches_object_type="",
+        matches_object_name="",
+        matches_attribute_name="",
+    ) -> list[str]:
+        keys = []
+        model = self["model"]
+        for key1 in model:
+            if matches_object_type.lower() != key1.lower():
+                continue
+            object_type = model[key1]
+            for key2, object_name in object_type.items():
+                if matches_object_name.lower() != key2.lower():
+                    continue
+                for key3 in object_name:
+                    if matches_attribute_name.lower() != key3.lower():
+                        continue
+                    attribute = object_name[key3]
+                    if isinstance(attribute, dict) and all(isinstance(x, datetime) for x in attribute.keys()):
+                        keys.append(f"model.{key1}.{key2}.{key3}")
```

Comment:
```suggestion
        for object_type, objects in model.items():
            if matches_object_type and matches_object_type.lower() != object_type.lower():
                continue
            for object_name, object_data in objects.items():
                if matches_object_name and matches_object_name.lower() != object_name.lower():
                    continue
                for attribute_name, attribute in object_data.items():
                    if matches_attribute_name and matches_attribute_name.lower() != attribute_name.lower():
                        continue
                    if isinstance(attribute, dict) and all(isinstance(x, datetime) for x in attribute.keys()):
                        keys.append(f"model.{object_type}.{object_name}.{attribute_name}")
```

Not tested! Just looks like the idea was to have optional matching attributes, so it seems we need to check whether a value has been passed in.

Reply:
Yes, it used to be there, not sure how it disappeared. Added back in 👍🏼